cmd/up.go

```diff
@@ -52,80 +63,254 @@ func up(cmd *cobra.Command, args []string) {
 			log.Printf("processing shipment: %v/%v", shipmentName, shipment.Env)
 		}
 
-		//iterate defined containers and apply container level updates
-		for _, container := range shipment.Containers {
+		//fetch the current state
+		shipmentObject := GetShipmentEnvironment(shipmentName, shipment.Env, token)
+
+		//creating a shipment is a different workflow than updating
+		//bulk create a shipment if it doesn't exist
+		if shipmentObject == nil {
 			if Verbose {
-				log.Printf("processing container: %v", container)
+				log.Println("shipment environment not found")
 			}
+			createShipment(shipmentName, dockerCompose, shipment, token)
+
+		} else {
+			//make changes to harbor based on compose files
+			updateShipment(shipmentObject, shipmentName, dockerCompose, shipment, token)
+
+			//TODO: desired state reconciliation
+		}
+
+		fmt.Println("done")
+
+	} //shipments
+}
+
+func createShipment(shipmentName string, dockerCompose DockerCompose, shipment ComposeShipment, token string) {
+
+	//map a ComposeShipment object (based on compose files) into
+	//a new NewShipmentEnvironment object
+
+	if Verbose {
+		log.Println("creating shipment environment")
+	}
+
+	//create object used to create a new shipment environment from scratch
+	newShipment := NewShipmentEnvironment{
+		Username: User,
+		Token:    token,
+		Info: NewShipmentInfo{
+			Name:  shipmentName,
+			Group: shipment.Group,
+		},
+	}
+
+	//add shipment-level env vars
+	newShipment.Info.Vars = make([]EnvVarPayload, 0)
+	newShipment.Info.Vars = append(newShipment.Info.Vars, envVar("CUSTOMER", shipment.Group))
+	newShipment.Info.Vars = append(newShipment.Info.Vars, envVar("PROPERTY", shipment.Property))
+	newShipment.Info.Vars = append(newShipment.Info.Vars, envVar("PROJECT", shipment.Project))
+	newShipment.Info.Vars = append(newShipment.Info.Vars, envVar("PRODUCT", shipment.Product))
+
+	//create environment
+	newShipment.Environment = NewEnvironment{
+		Name: shipment.Env,
+		Vars: make([]EnvVarPayload, 0),
+	}
 
-			//lookup the container in the list of services in the docker-compose file
-			dockerService := dockerCompose.Services[container]
+	//add environment-level env vars
+	for name, value := range shipment.Environment {
+		newShipment.Environment.Vars = append(newShipment.Environment.Vars, envVar(name, value))
+	}
+
+	//add barge envvar (note that this is being moved to the provider level soon)
+	if shipment.Barge == "" {
```

Comment:
it was pushed today. So now trigger reads from the provider barge value. Plus shipit won't allow you to create a new shipment without a barge value on the provider.